src/main/java/seedu/address/model/parcel/Parcel.java

```diff
@@ -192,6 +192,12 @@ public int compareTo(Object o) {
         Parcel other = (Parcel) o;
         if (other == this) {
             return 0;
+        } else if (this.getDeliveryDate().compareTo(other.getDeliveryDate()) == 0) {
```

Comment:
add comments for your if else statements please, its too many if else, so comments would be good to indicate  how you are doing your comparisons